ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/NonStakeholderInformeesITSpec.scala

```diff
@@ -0,0 +1,174 @@
+          template = "Foo1",
+          weight = 1,
+          payloadSizeBytes = 0,
+        )
+      ),
+      nonConsumingExercises = None,
+      consumingExercises = None,
+    )
+    for {
+      ledgerApiServicesF <- LedgerApiServices.forChannel(
+        channel = channel,
+        authorizationHelper = None,
+      )
+      apiServices: LedgerApiServices = ledgerApiServicesF("someUser")
+      submitter = CommandSubmitter(
+        names = new Names(),
+        benchtoolUserServices = apiServices,
+        adminServices = apiServices,
+        metricRegistry = new MetricRegistry,
+        metricsManager = NoOpMetricsManager(),
+      )
+      (signatory, observers, divulgees) <- submitter.prepare(submissionConfig)
+      tested = new FooSubmission(
+        submitter = submitter,
+        maxInFlightCommands = 1,
+        submissionBatchSize = 5,
+        submissionConfig = submissionConfig,
+        signatory = signatory,
+        allObservers = observers,
+        allDivulgees = divulgees,
+      )
+      _ <- tested.performSubmission()
+      (treeResults_divulgee0, flatResults_divulgee0) <- observeAllTemplatesForParty(
+        party = divulgees(0),
+        apiServices = apiServices,
+        expectedTemplateNames = expectedTemplateNames,
+      )
+      (treeResults_divulgee1, flatResults_divulgee1) <- observeAllTemplatesForParty(
+        party = divulgees(1),
+        apiServices = apiServices,
+        expectedTemplateNames = expectedTemplateNames,
+      )
+      (treeResults_observer0, flatResults_observer0) <- observeAllTemplatesForParty(
+        party = observers(0),
+        apiServices = apiServices,
+        expectedTemplateNames = expectedTemplateNames,
+      )
+      (treeResults_signatory, _) <- observeAllTemplatesForParty(
+        party = signatory,
+        apiServices = apiServices,
+        expectedTemplateNames = expectedTemplateNames,
+      )
+    } yield {
+      // Creates of Foo1 are divulged to "divulgee" party,
+      // thus, they are visible on transaction trees stream but absent from flat transactions stream.
+      {
+        // Divulge0
+        val treeFoo1 = treeResults_divulgee0.numberOfCreatesPerTemplateName("Foo1")
+        val flatFoo1 = flatResults_divulgee0.numberOfCreatesPerTemplateName("Foo1")
+        treeFoo1 shouldBe 100 withClue ("number of Foo1 contracts visible to divulgee0 on tree transactions stream")
+        flatFoo1 shouldBe 0 withClue ("number of Foo1 contracts visible to divulgee0 on flat transactions stream")
+        val divulger = treeResults_divulgee0.numberOfCreatesPerTemplateName("Divulger")
+        divulger shouldBe 4 withClue ("number divulger contracts visible to divulgee0")
+      }
+      {
+        // Divulgee1
+        val treeFoo1 = treeResults_divulgee1.numberOfCreatesPerTemplateName("Foo1")
+        val flatFoo1 = flatResults_divulgee1.numberOfCreatesPerTemplateName("Foo1")
+        treeFoo1 shouldBe 10 +- 9
```

Comment:
This 10 is a bit of a magic number.  Sometime doing the calculation longhand can add clarity:
`shouldBe (numberOfSubmissions/10) +- ...`

Also because this is a statistical measure there is no guarantee that there will not be in excess of +- 9

Reply:
Changing to `treeFoo1 should be > 0`.
Because divulgee1 has a 10% chance of being chosen, it's chance of never be chosen of 100 instances is 2.7 in 100k:
```
$ .9**100
2.6561398887587544e-05

$ 1/(0.9**100)
37648.61949599017
```
So this test will fail once every 37648 executions.